File: app/src/main/cpp/hongbao/hongbao.cpp
//
// Created by yangyu on 2019-03-18.
//
#include <jni.h>
#include <string>
#include <stdlib.h>

#include "hongbao.h"   //引入hongbao.h头文件

//HongBao.java中native方法的实现体
jint add(JNIEnv *env, jclass clazz, jint x, jint y) {
    return x + y;
}

//HongBao.java中的参数映射表
static JNINativeMethod gMethods_HongBao[] = {
        {"add", "(II)I", (void *) add},
};
//注册 HongBao 中的native方法
int registerNatives_HongBao(JNIEnv *engv) {
    jclass clazz;
    clazz = engv->FindClass("com/example/myapplication/HongBao");
    if (clazz == NULL) {
        return JNI_FALSE;
    }
    if (engv->RegisterNatives(clazz, gMethods_HongBao,sizeof(gMethods_HongBao) / sizeof(gMethods_HongBao[0])) < 0) {
        return JNI_FALSE;
    }
    return JNI_TRUE;
}

File: app/src/main/java/com/example/myapplication/HongBao.java
package com.example.myapplication;

public class HongBao {
    /**
     * 假设这个类有很多方法，我们省略一些其他的方法...
     */
    /**
     * 该jni方法是测试两个数字相加并返回该结果...
     * @param x
     * @param y
     * @return
     */
    public static native int add(int x, int y);

    /**
     * 测试两个整形数字相减
     * @param x
     * @param y
     * @return
     */
    public static int sub(int x, int y) {
        return x - y;
    }
}
